src/diffusers/models/unet_2d_condition_flax.py

```diff
@@ -0,0 +1,207 @@
+from typing import Tuple
+
+import flax.linen as nn
+import jax
+import jax.numpy as jnp
+# from flax.core.frozen_dict import FrozenDict
+
+from ..configuration_utils import ConfigMixin, flax_register_to_config
+from ..modeling_flax_utils import FlaxModelMixin
+from .embeddings_flax import FlaxTimestepEmbedding, FlaxTimesteps
+from .unet_blocks_flax import (
+    FlaxDownBlock2D,
+    FlaxCrossAttnDownBlock2D,
+    FlaxUNetMidBlock2DCrossAttn,
+    FlaxUpBlock2D,
+    FlaxCrossAttnUpBlock2D,
+)
+
+
+@flax_register_to_config
+class FlaxUNet2DConditionModel(nn.Module, FlaxModelMixin, ConfigMixin):
+    sample_size=32
+    in_channels=4
+    out_channels=4
+    down_block_types=("CrossAttnDownBlock2D", "CrossAttnDownBlock2D", "CrossAttnDownBlock2D", "DownBlock2D")
+    up_block_types=("UpBlock2D", "CrossAttnUpBlock2D", "CrossAttnUpBlock2D", "CrossAttnUpBlock2D")
+    block_out_channels=(224, 448, 672, 896)
+    layers_per_block=2
+    attention_head_dim=8
+    cross_attention_dim=768
+    dropout=0.1
+    dtype: jnp.dtype = jnp.float32
```

Comment:
>> I'm having a strange bug where the configuration is not correctly applied. The json file is correctly read in from_pretrained, but the signature of the __init__ method does not contain all the parameters by the time it's inspected in extract_init_dict

flax being flax, you need to provide typings

```suggestion
class FlaxUNet2DConditionModel(nn.Module, FlaxModelMixin, ConfigMixin):
    sample_size:int=32
    in_channels:int=4
    out_channels:int=4
    down_block_types:Tuple=("CrossAttnDownBlock2D", "CrossAttnDownBlock2D", "CrossAttnDownBlock2D", "DownBlock2D")
    up_block_types:Tuple=("UpBlock2D", "CrossAttnUpBlock2D", "CrossAttnUpBlock2D", "CrossAttnUpBlock2D")
    block_out_channels:Tuple=(224, 448, 672, 896)
    layers_per_block:int=2
    attention_head_dim:int=8
    cross_attention_dim:int=768
    dropout:float=0.1
    dtype: jnp.dtype = jnp.float32
```

Reply:
Of course 🤦 ! I'll try that, thanks a lot!

Reply:
plese let me know if it fixe the issue, I haven't tested locally

Reply:
It worked like a charm, thanks again!